File: src/CustomSDK/05_Examples/Python/CustomModules/CustomModuleExecutable4Python/src/CustomModuleExecutable.py
from CustomFramework4Python import CustomFrameworkManager

import os
import sys
import time
import argparse

class CustomModuleExecutable:
  def doSomething(self):
    pass

def main():
  parser = argparse.ArgumentParser()
  parser.add_argument("-c", "--config_path", type=str, required=True, help="Path to the configuration file")
  
  args = parser.parse_args()
  if not hasattr(args, "-h"):
    config_path = args.config_path
    
    custom_framework_manager = CustomFrameworkManager.instance()
    custom_framework_manager.initializeWithConfig(config_path)
    
    custom_module_executable = CustomModuleExecutable()
    custom_module_executable.doSomething()
  else:
    parser.print_help()

if __name__ == "__main__":
  main()

File: src/CustomSDK/05_Examples/Java/CustomModules/CustomModuleExecutable4Java/src/examples/CustomModuleExecutable.java
package examples;

import CustomFramework4Java.CustomFrameworkManager;
import CustomFramework4Java.AbstractCustomModule;
import CustomFramework4Java.LogManager;

import org.apache.commons.cli.*;

public class CustomModuleExecutable
{
  void doSomething()
  {
  }

  public static void main(String[] args)
  {
    Options options = new Options();

    Option option_config_path = new Option("c", "config_path", true, "Path to config file");
    option_config_path.setRequired(true);

    Option option_help = new Option("h", "help", false, "Show help");

    options.addOption(option_config_path);
    options.addOption(option_help);

    try
    {
      CommandLineParser parser = new DefaultParser();
      CommandLine command_line = parser.parse(options, args);
      String config_path = command_line.getOptionValue("config_path");
      try
      {
        if(CustomFrameworkManager.instance().initializeWithConfig(config_path))
        {
          CustomModuleExecutable custom_module_executable = new CustomModuleExecutable();
          custom_module_executable.doSomething();
        }
      }
      catch(Exception e)
      {
        e.printStackTrace();
      }
    }
    catch (ParseException e)
    {
      HelpFormatter formatter = new HelpFormatter();
      System.out.println(e.getMessage());
      formatter.printHelp("Allowed options", options);
    }
  }
}
